### PR TITLE
Reject mocks

### PR DESCRIPTION
**Pull request by hally9k**
Hey, thanks for your work on this.
I have just put the most naive implementation of mocking failed requests here.
More to start the conversation than anything.
This does serve my current use case though.
Any thoughts on how we could make this better?

### CODE REVIEW

**Comment by jefflau** [comment]
Hey. Sorry for the slow reply to this. I thought I saw it closed about an hour after you sent this and I didn't realised it had been reopened (Maybe I'm mistaken?). Anyway this looks interesting. Could you maybe add an example in the `README.md` so I can see how you're using it?

**Comment by hally9k** [comment]
@jefflau I've added a readme example. The obvious iteration on this would be to be able to pass a rejection response into `mockReject()` but I'm keen to hear your thoughts...

**Comment by Si1kIfY** [comment]
Rejecting the response with status codes would be good but that would require changing `ResponseWrapper`

I can do that in a separate PR once this is implemented

File: src/index.js [hunk 1]
```diff
@@ -43,12 +43,24 @@ fetch.mockResponse = (body, init) => {
   );
 };
 
+fetch.mockReject = (body, init) => {
```
Comment (Si1kIfY) [inline src/index.js]: Do you need to pass body, init into `mockReject` when they are not used? (same for `MockRejectOnce`)

Comment (hally9k) [inline src/index.js]: This is weird, these lines are not in my PR... Keen to hear how to cleanly introduce a good rejection response. I don't know the fetch spec well enough to come off the top for that...

**Comment by jefflau** [comment]
I haven't got as much time to maintain this module. I think rejects would be a great addition, so I'm going to merge this and see how you guys develop this part further. Thank you for putting the time to make and review this pull request

**Comment by hally9k** [comment]
@Si1kIfY Really keen to see how you can iterate on this to allow the rejection response to be defined...

**Pull request merged**